### PR TITLE
Use node-portfinder to avoid error when port is used

### LINKED ISSUE #320: use portfinder to avoid error when port is used
https://github.com/indexzero/node-portfinder

### PR DESCRIPTION
**Pull request by thangngoc89**
Close #320

### CODE REVIEW

File: src/builder/server.js [hunk 1]
```diff
@@ -196,17 +197,30 @@ export default (options = {}) => {
...
-      opn(href.replace(devHost, "localhost"))
+
+    if (port !== devHost) {
+      log(`Port ${ devPort } is not available. Using port ${ port } instead.`)
     }
+    server.listen(port, devHost, (err) => {
+      if (err) {
+        log(err)
+
+        return
```
Comment (MoOx) [inline src/builder/server.js]: why not throw?

**Comment by MoOx** [comment]
May be we should add integration test for this. Cause coverage is decreasing.

File: src/builder/server.js (hunk 1 above)
Comment (thangngoc89) [inline src/builder/server.js]: It was there before. You like throw. I'll do it.

**Comment by thangngoc89** [comment]
> May be we should add integration test for this. 

Sure. On it

**Comment by thangngoc89** [comment]
> Cause coverage is decreasing.

How do you know about that ?

**Comment by MoOx** [comment]
> How do you know about that ? 

You add code that is not tested :)

File: __tests__/cli.js [hunk 2]
```diff
@@ -38,10 +38,38 @@ test.cb("should NOT throw if a CLI flag is recognized", (t) => {
   )
 
   // ...or be ok quickly
-  // (so we assume it's ok and kill the process, we don't need the actual build)
+  // (so we assume it"s ok and kill the process, we don"t need the actual build)
```
Comment (MoOx) [inline __tests__/cli.js]: ‘ (alt + ')

File: __tests__/cli.js [hunk 3]
```diff
@@ -38,10 +38,38 @@ test.cb("should NOT throw if a CLI flag is recognized", (t) => {
...
   const timeout = setTimeout(() => {
     t.pass()
     t.end()
     child.kill()
   }, 500)
 })
+
+test.cb("should NOT throw if port is used", (t) => {
```
Comment (MoOx) [inline __tests__/cli.js]: should, not "should NOT" ?

**Comment by thangngoc89** [comment]
@MoOx integration-tests was not run

File: __tests__/cli.js (hunk 3 above)
Comment (MoOx) [inline __tests__/cli.js]: hum ok nvm.

File: __tests__/cli.js [hunk 4]
```diff
@@ -38,10 +38,38 @@ test.cb("should NOT throw if a CLI flag is recognized", (t) => {
...
+          t.end()
+        }
+      }
+    )
+
+    const timeout = setTimeout(() => {
+      t.pass()
+      t.end()
+      child.kill()
+    }, 500)
```
Comment (MoOx) [inline __tests__/cli.js]: is 500ms enough? Did you check without your added code in server.js?

Comment (thangngoc89) [inline __tests__/cli.js]: Yeah. You're right. The tests weren't killed.

**Comment by thangngoc89** [comment]
@MoOx can you have a look at the integration-tests ? original tests in that file aren't killed.

File: __tests__/cli.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ import { exec } from "child_process"
 const asyncExec = asyncify(exec)
 
 const target = join(__dirname, "..", "test-boilerplate")
-const execOpts = { cwd: target }
+const execOpts = { cwd: target, stdio: "inherit" }
```
Comment (MoOx) [inline __tests__/cli.js]: please don't do that. This can pollute test output.

**Comment by thangngoc89** [comment]
Blocked by #346

**Comment by MoOx** [comment]
wat https://travis-ci.org/MoOx/statinamic/jobs/120807447#L625
Any idea @thangngoc89?

**Comment by thangngoc89** [comment]
No idea. I'll take a look

**Comment by thangngoc89** [comment]
All tests passed locally

**Comment by MoOx** [comment]
Same here, local tests ok.

**Comment by thangngoc89** [comment]
I can't print the log even with inherit

**Comment by MoOx** [comment]
woot. 5432 seems to be used by travis, I tried with 8081, and tests are ok on travis.

**Comment by MoOx** [comment]
I cleaned this PR, waiting for tests.

**Comment by MoOx** [comment]
FYI, that may be the reason :)
screen shot 2016-04-09 at 14 47 39

**Comment by thangngoc89** [comment]
Haha. Nice.

**Pull request merged**